dash/development/component_generator.py

```diff
@@ -10,16 +10,34 @@
 
 import pkg_resources
 
-from ._py_components_generation import generate_class_file
-from ._py_components_generation import generate_imports
-from ._py_components_generation import generate_classes_files
+from ._py_components_generation import generate_class_file, generate_imports
 
 
 class _CombinedFormatter(argparse.ArgumentDefaultsHelpFormatter,
                          argparse.RawDescriptionHelpFormatter):
     pass
 
 
+def generate_component_files(project_shortname, metadata, *component_generators):
+    components = []
+    for component_path, component_data in metadata.items():
+        component_name = component_path.split('/')[-1].split('.')[0]
+        components.append(component_name)
+
+        for generator in component_generators:
+            generator(
+                component_name,
+                component_data['props'],
+                component_data['description'],
+                project_shortname
+            )
+
+    return components
+
+def generate_suite_files(project_shortname, components, metadata, *suite_generators):
```

Comment:
I'm pushing the writing of `metadata.json` into this function here, as the path will depend on the language.